src/public/media.ts

```diff
@@ -77,3 +92,265 @@ export function captureImage(callback: (error: SdkError, files: File[]) => void)
   const messageId = sendMessageRequestToParent('captureImage');
   GlobalVars.callbacks[messageId] = callback;
 }
+
+/**
+ * Media object returned by the select Media API
+ */
+export class Media extends File {
+  /**
+   * A preview of the file which is a lightweight representation.
+   * In case of images this will be a thumbnail/compressed image in base64 encoding.
+   */
+  public preview: string;
+
+  /**
+   * Gets the media in chunks irrespecitve of size, these chunks are assembled and sent back to the webapp as file/blob
+   * @param callback returns blob of media
+   */
+  public getMedia(callback: (error: SdkError, blob: Blob) => void): void {
+    if (!callback) {
+      throw new Error('[get Media] Callback cannot be null');
+    }
+    ensureInitialized(FrameContexts.content, FrameContexts.task);
+    if (!isAPISupportedByPlatform(mediaAPIVersion)) {
+      const oldPlatformError: SdkError = { errorCode: ErrorCode.OLD_PLATFORM };
+      callback(oldPlatformError, null);
+      return;
+    } else if (!validGetMediaInputs(this.mimeType, this.format, this.content)) {
+      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      callback(invalidInput, null);
+      return;
+    }
+    let actionName = generateGUID();
+    let helper: MediaHelper = {
+      mediaMimeType: this.mimeType,
+      assembleAttachment: [],
+    };
+    const params = [actionName, this.content];
+    this.content && callback && sendMessageRequestToParent('getMedia', params);
+    function handleGetMediaRequest(response: string): void {
+      if (callback) {
+        let mediaResult: MediaResult = JSON.parse(response);
+        if (mediaResult.error) {
+          callback(mediaResult.error, null);
+        } else {
+          if (mediaResult.mediaChunk) {
+            if (mediaResult.mediaChunk.chunkSequence <= 0) {
```

Comment:
Why are there two cases here? Do we have two different ways of transmitting the files across? Please add a comment explaining that.

Reply:
added